tests/public.rs

```diff
@@ -0,0 +1,101 @@
+/// Check if needed fields are still public.
+
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+extern crate mp4parse as mp4;
+
+use std::io::{Cursor, Read};
+use std::fs::File;
+
+// Took from https://github.com/GuillaumeGomez/audio-video-metadata/blob/9dff40f565af71d5502e03a2e78ae63df95cfd40/src/metadata.rs#L53
+//
+// Not important if it fails or not. If it compiles, it means it's all good.
+#[test]
+fn check_mp4() {
+    let buf = if let Some(mut fd) = File::open("examples/minimal.mp4").ok() {
+        let mut buf = Vec::new();
+
+        match fd.read_to_end(&mut buf) {
+            Ok(_) => buf,
+            Err(_) => panic!("File error"),
+        }
+    } else {
+        panic!("Unknown file")
```

Comment:
I'd rather have this up near the `File::open` and save a level of indent. For example you could use a pair of `match` stanzas like in `examples/dump.rs`. I think the readability is more important than freeing the file object sooner.

Using .expect() would be appropriate in a test too.